### PR TITLE
Helmchart

### PR DESCRIPTION
**Pull request by YonatanItzhak**
Please review this and let me know if anything needs to change

### CODE REVIEW

File: maskshop-chart/templates/api-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  selector:
+    matchLabels:
+      app: api
+  template:
+    metadata:
+      labels:
+        app: api
+    spec:
+      containers:
+      - image: otomato/maskshop-api:0.1
```
Comment (antweiss) [inline maskshop-chart/templates/api-deployment.yaml]: Please define image name and tag in values.yaml as 
ApiImageName and ApiImageTag

File: maskshop-chart/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+version: 0.1.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application.
+appVersion: 1.16.0
```
Comment (antweiss) [inline maskshop-chart/Chart.yaml]: This version should currently be 0.1

File: maskshop-chart/templates/front-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  selector:
+    matchLabels:
+      app: front
+  template:
+    metadata:
+      labels:
+        app: front
+    spec:
+      containers:
+      - image: otomato/maskshop-front:0.1
```
Comment (antweiss) [inline maskshop-chart/templates/front-deployment.yaml]: Same - image name and tag should be in values

File: maskshop-chart/templates/mongo-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  selector:
+    matchLabels:
+      app: mongo
+  template:
+    metadata:
+      labels:
+        app: mongo
+    spec:
+      containers:
+      - image: mongo
```
Comment (antweiss) [inline maskshop-chart/templates/mongo-deployment.yaml]: same here - it's best not to deploy anything as latest - find out what current mongo version is and use that as tag

File: maskshop-chart/values.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
+# Default values for maskshop-chart.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: nginx
```
Comment (antweiss) [inline maskshop-chart/values.yaml]: This needs to be replaced with actual image names and tags. We aren't deploying nginx

File: maskshop-chart/values.yaml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  pullPolicy: IfNotPresent
+
+imagePullSecrets: []
+nameOverride: ""
+fullnameOverride: ""
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
```
Comment (antweiss) [inline maskshop-chart/values.yaml]: serviceAccount isn't used - it isn;t needed here

File: maskshop-chart/templates/mongo-deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+mongoVersion: apps/v1
```
Comment (antweiss) [inline maskshop-chart/templates/mongo-deployment.yaml]: Just tested. This prevents it from installing. Should be `apiVersion`. Just an unfortunate replace I suppose ;)

Comment (YonatanItzhak) [reply maskshop-chart/templates/mongo-deployment.yaml]: Yes haha

File: maskshop-chart/templates/api-service.yaml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+kind: Service
+metadata:
+  labels:
+    app: api
+  name: api
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 8080
```
Comment (antweiss) [inline maskshop-chart/templates/api-service.yaml]: this should be 80

File: maskshop-chart/templates/front-service.yaml [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+kind: Service
+metadata:
+  labels:
+    app: front
+  name: front
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 8080
```
Comment (antweiss) [inline maskshop-chart/templates/front-service.yaml]: this should be 80

File: maskshop-chart/templates/front-deployment.yaml (hunk 3 above)
Comment (antweiss) [reply maskshop-chart/templates/front-deployment.yaml]: Please also add env variable for this container:
`
 env:
    - name: INITDB
      value: "true"
`
See here for more details: https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/

File: maskshop-chart/values.yaml [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+securityContext: {}
+  # capabilities:
+  #   drop:
+  #   - ALL
+  # readOnlyRootFilesystem: true
+  # runAsNonRoot: true
+  # runAsUser: 1000
+
+service:
```
Comment (antweiss) [inline maskshop-chart/values.yaml]: the defines service type and port aren't used anywhere - this is confusing
either define values for all services or remove this from the values.yaml

File: maskshop-chart/values.yaml [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+  type: ClusterIP
+  port: 80
+
+ingress:
```
Comment (antweiss) [inline maskshop-chart/values.yaml]: we're not using ingress so this is also unnecessary

**Pull request merged**